### PR TITLE
Add Contribution Guidelines

### LINKED ISSUE #122: Docs: Add Developent.md
describe whitch things are stored wher, what is wat & how to start programming

FAQ for Devs:
 * SemanticVersioning
 * Translation -> Crowdin
 * ...

### PR DESCRIPTION
**Pull request by 6543**
close #122

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+included in the next released version.
+
+
+## Building
+
+See the [Readme](README.md)
+
+## Styleguide
+
+Imports are sorted, src contain frontend only, src-tauri contain backend only.
```
Comment (crapStone) [inline CONTRIBUTING.md:45]: we should specify this part more

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+## Release Cycle
+
+Before v1.0 there will be a release if needed.
+After v1.0 one release halve a year (bugfixes excluded).
+
+We use semantic versioning.
```
Comment (crapStone) [inline CONTRIBUTING.md:52]:
```suggestion
We use [semantic](https://semver.org/) versioning.
```

**Review by crapStone — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (6543) [reply CONTRIBUTING.md:45]: Suggestions are welcome ;)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+included in the next released version.
+
+
+## Building
+
+See the [Readme](README.md)
+
+## Styleguide
+
+Imports are sorted, src contain frontend only, src-tauri contain backend only.
```
Comment (crapStone) [inline CONTRIBUTING.md:45]:
```suggestion
## Frontend/Angular

- all frontend-related code is in `src` folder
- imports should be arranged like this with empty lines in between:
  1. external libraries/browser builtins
  2. own modules
- imports should be sorted alphabetically
- use style defined in prettierrc:
  - always use trailing commas
  - use singlequotes
  - use 2 spaces to indent code
  - ...

## Backend/Rust

- all backend-ralated code is in `src-tauri`
- imports should be arranged like this with empty lines in between:
  1. standard library (`std`)
  2. external librarys
  3. internal libraries (prefixed with `cabr2_`)
  4. intra-library imports
- use style defined in `rustfmt.toml`:
  - always use trailing commas
  - use 2 spaces to indent code
  - ...
```

**Review by crapStone — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 3 above)
Comment (crapStone) [reply CONTRIBUTING.md:45]: @Epsilon02 something to add?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+A SATISFACTORY PERCENTAGE it will be synced back into this repo and
+included in the next released version.
+
+
+## Building
+
+See the [Readme](README.md)
+
+## Styleguide
+
```
Comment (6543) [inline CONTRIBUTING.md:44]:
```suggestion
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+- all frontend-related code is in `src` folder
+- imports should be arranged like this with empty lines in between:
+  1. external libraries/browser builtins
+  2. own modules
+- imports should be sorted alphabetically
+- use style defined in prettierrc:
+  - always use trailing commas
+  - use singlequotes
+  - use 2 spaces to indent code
+  - ...
```
Comment (6543) [inline CONTRIBUTING.md:56]: @Epsilon02 something to add?

File: CONTRIBUTING.md (hunk 3 above)
Comment (Epsilon02) [reply CONTRIBUTING.md]: Lgtm

File: CONTRIBUTING.md (hunk 5 above)
Comment (Epsilon02) [reply CONTRIBUTING.md:54]: Lgtm

Comment (6543) [reply CONTRIBUTING.md:54]: @Epsilon02 I think you have to use the review function :D

**Review by Epsilon02 — APPROVED** [review]

**Review by crapStone — APPROVED** [review]

**Pull request merged**